File: frontend/src/types/subscription.ts
export interface Plan {
  id: string;
  name: string;
  slug: string;
  monthly_price_cents: number;
  max_leads_per_month: number;
  max_users: number;
  max_extractions_per_day: number;
}

export interface Subscription {
  id: string;
  plan: Plan;
  status: string;
  current_period_start: string | null;
  current_period_end: string | null;
  created_at: string;
}

export interface Usage {
  leads_extracted: number;
  leads_exported: number;
  max_leads_per_month: number;
  usage_percentage: number;
}

File: backend/app/schemas/subscription.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class PlanResponse(BaseModel):
    id: uuid.UUID
    name: str
    slug: str
    monthly_price_cents: int
    max_leads_per_month: int
    max_users: int
    max_extractions_per_day: int

    model_config = {"from_attributes": True}


class SubscriptionResponse(BaseModel):
    id: uuid.UUID
    plan: PlanResponse
    status: str
    current_period_start: datetime | None
    current_period_end: datetime | None
    created_at: datetime

    model_config = {"from_attributes": True}


class UsageResponse(BaseModel):
    leads_extracted: int = 0
    leads_exported: int = 0
    max_leads_per_month: int = 0
    usage_percentage: float = 0.0


class CheckoutRequest(BaseModel):
    plan_slug: str
    success_url: str
    cancel_url: str


class CheckoutResponse(BaseModel):
    checkout_url: str
